src/renderer/renderScene.ts

```diff
@@ -32,9 +36,20 @@ export function renderScene(
     renderSelection?: boolean;
   } = {},
 ) {
-  if (!canvas) return false;
+  // Use offsets insteads of scrolls if available
```

Comment:
Just reordered because I feel like merging params with overwrites should be the first thing done in a function.

Reply:
We want to bail as early as possible when no work needs to be done, otherwise it’s just wasted.

In this case, we shouldn’t even come here if canvas is not created!

Reply:
Yes, right, I don't consider params merging as an expensive task, but I will reorder :+1:

Reply:
Done.